packages/core-tabs/core-tabs.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import coreTabs from './core-tabs'
+import {exclude} from '../utils'
+
+const DEFAULTS = {openIndex: null, onToggle: null}
+
+export default class Tabs extends React.Component {
+  componentDidMount () {
+    coreTabs(this.el.firstElementChild) // Mount client side only to avoid rerender
+    this.el.addEventListener('tabs.toggle', this.props.onToggle)
+  }
+  componentDidUpdate () { coreTabs(this.el.firstElementChild) } // In case content changes
+  componentWillUnmount () { this.el.removeEventListener('tabs.toggle', this.props.onToggle) }
+  render () {
+    const open = this.props.open || 0
+    const attr = exclude(this.props, DEFAULTS, {ref: (el) => (this.el = el)})
+
+    return React.createElement('div', attr,
```

Comment:
Could we just write this as jsx? Might be easier to read?  (Although we don't do it for the other components.) Anyways, I don't have strong feelings one way or the other, so not necessary.

Reply:
Not really unfortunately - we need the fact that `React.cloneElement` copys `ref` and `key` under the hood, so it will be like this for now :)